src/extension/background-script/actions/nostr/generatePrivateKey.ts

```diff
@@ -22,16 +23,19 @@ const generatePrivateKey = async (message: MessageNostrPrivateKeyGenerate) => {
       };
     }
     const mnemonic = new Mnemonic(decryptData(account.mnemonic, password));
-    const nostrPrivateKey = mnemonic.deriveNostrPrivateKeyHex();
-
-    return {
-      data: nostrPrivateKey,
-    };
+    nostrPrivateKey = mnemonic.deriveNostrPrivateKeyHex();
+  } else if (message.args?.mnemonic) {
```

Comment:
I'm just wondering if this is the best approach since we add more complexity to these functions to both handle accounts and non-accounts.

But I can't see any other way except actually creating the account before we validate it :thinking: